starter-code/src/functions-and-arrays.js

```diff
@@ -98,3 +193,53 @@ var matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+function greatestProduct (matrix){
+  var maxProduct = 0;
+  for(i = 0;  i === 19; i++){
+    for(j = 0; j === 19; j++){
+      var productLeft = matrix[i][j];
+      var productRight = matrix[i][j];
+      var productUp = matrix[i][j];
+      var productDown = matrix[i][j];
+      var maxProductIJ = 0;
+      for (k = 1; k === 3; k++){
+        if (i-k < 0){
+          productUp = productUp * 1;
+        }else{
+          productUp *= matrix[i-k][j];
+        }
+      }
+      for(k = 1; k === 3; k++){
+        if (i + k > 19){
+          productDown = productDown * 1;
+        }else{
+          productDown *= matrix[i+k][j];
+        }
+      }
+      for (k = 1; k === 3; k++){
+        if (j-k < 0){
+          productLeft = productLeft * 1;
+        }else{
+          productLeft *= matrix[i][j-k];
+        }
+      }
+      for(k = 1; k === 3; k++){
+        if (j + k > 19){
+          productRight = productRight * 1;
+        }else{
+          productRight *= matrix[i][j+k];
+        }
+      }
```

Comment:
Diría que te sobran 3 `for`, de todas maneras recuerda que el producto de izquierda a derecha y de derecha a izquierda o el producto de arriba a abajo y de abajo a arriba dan el mismo resultado, así que seguro podrías quitarte 2 bloques `for`